pallets/liquidity-pools/src/inbound.rs

```diff
@@ -173,80 +200,165 @@ impl<T: Config> Pallet<T> {
+			pool_id,
+			tranche_id,
+			investor: investor.clone().into(),
+			currency: currency_index.index,
+			tranche_tokens_payout,
+		};
 
-		// TODO(subsequent PR): Handle response `ExecutedDecreaseRedemption` message to
-		// source destination which should refund the decreased amount. This includes
-		// transferring the amount from the investor to the domain locator account of
-		// the origination domain.
-		//
-		// NOTES:
-		// 	* Blocked by https://github.com/centrifuge/centrifuge-chain/pull/1363
-		// 	* Should be handled by `pallet-foreign-investments`
+		T::OutboundQueue::submit(investor, destination.domain(), message)?;
 
 		Ok(())
 	}
 
+	/// Cancels an existing redemption order of the investor by decreasing the
+	/// redemption by the entire unprocessed amount.
+	///
+	/// Initiates a return `ExecutedDecreaseRedemption` message to refund the
+	/// decreased amount on the source domain.
+	pub fn handle_cancel_redeem_order(
+		pool_id: T::PoolId,
+		tranche_id: T::TrancheId,
+		investor: T::AccountId,
+		currency_index: GeneralCurrencyIndexOf<T>,
+		destination: DomainAddress,
+	) -> DispatchResult {
+		let invest_id: T::TrancheCurrency = Self::derive_invest_id(pool_id, tranche_id)?;
+		let amount = T::ForeignInvestment::redemption(&investor, invest_id)?;
+		Self::handle_decrease_redeem_order(
+			pool_id,
+			tranche_id,
+			investor,
+			currency_index,
+			amount,
+			destination,
+		)
+	}
+
 	/// Collect the results of a user's invest orders for the given investment
 	/// id. If any amounts are not fulfilled, they are directly appended to the
 	/// next active order for this investment.
+	///
+	/// Transfers collected amount from investor's sovereign account to the
+	/// sending domain locator.
+	///
+	/// NOTE: In contrast to collecting a redemption, investments can be
+	/// collected entirely synchronously as it does not involve swapping. It
+	/// simply transfers the tranche tokens from the pool to the sovereign
+	/// investor account on the local domain.
 	pub fn handle_collect_investment(
 		pool_id: T::PoolId,
 		tranche_id: T::TrancheId,
 		investor: T::AccountId,
+		currency_index: GeneralCurrencyIndexOf<T>,
+		destination: DomainAddress,
 	) -> DispatchResult {
 		let invest_id: T::TrancheCurrency = Self::derive_invest_id(pool_id, tranche_id)?;
+		let currency_index_u128 = currency_index.index;
+		let payment_currency = Self::try_get_payment_currency(invest_id.clone(), currency_index)?;
+		let pool_currency =
+			T::PoolInspect::currency_for(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+
+		let ExecutedForeignCollectInvest::<T::Balance> {
+			amount_currency_payout,
+			amount_tranche_tokens_payout,
+		} = T::ForeignInvestment::collect_foreign_investment(
+			&investor,
+			invest_id.clone(),
+			payment_currency,
+			pool_currency,
+		)?;
+
+		T::Tokens::transfer(
+			invest_id.into(),
+			&investor,
+			&Domain::convert(destination.domain()),
+			amount_tranche_tokens_payout,
+			false,
+		)?;
```

Comment:
Q: Should this rather be part of `collect_foreign_investment`?

Reply:
Update: Should not based on [this comment](https://github.com/centrifuge/centrifuge-chain/pull/1473#discussion_r1323198290).